app/lib/actions/pulp3/repository/remove_units.rb

```diff
@@ -0,0 +1,21 @@
+module Actions
+  module Pulp3
+    module Repository
+      class RemoveUnits < Pulp3::AbstractAsyncTask
+        def plan(repository, smart_proxy, options)
+          plan_self(repository_id: repository.id,
+                    smart_proxy_id: smart_proxy.id,
+                    options: options)
+        end
+
+        def invoke_external_task
+          repo = ::Katello::Repository.find(input[:repository_id])
+          content_unit_ids = input[:options][:contents]
+          file_units = ::Katello::RepositoryFile.find(content_unit_ids)
```

Comment:
I think we need to narrow this down with repository. So the query would be something like:
` Katello::RepositoryFile.where(:file_id => content_unit_ids :repository_id => repo.id)`